app.py

```diff
@@ -46,10 +47,10 @@ def _list_pulp_python_repositories(instance_host: str, *, username: str, passwor
     response.raise_for_status()
 
     for result in response.json()["results"]:
-        simple_repository = (
-            f"{result['base_url']}simple" if result["base_url"].endswith("/") else f"{result['base_url']}/simple"
+        path = urlparse(result["base_url"]).path
+        yield urlunparse(
+            ("https", instance_host, f"{path}/simple" if not path.endswith("/") else f"{path}simple", "", "", "")
```

Comment:
`instance_host` should be also slash validated too shouldn't it?
```suggestion
            ("https", f"{instance_host}/" if not instance_host.endswith("/") else f"{instance_host}", f"{path}/simple" if not path.endswith("/") else f"{path}simple", "", "", "")
```